text/build_process.md

```diff
@@ -0,0 +1,122 @@
+
+* Rust ups will still have to be meticulously scheduled.
+
+#### Use Semantic Versioning
+
+Cargo [supports SemVer](http://doc.crates.io/crates-io.html#using-cratesio-based-crates). This less permissive approach would allow us to be explicit on when our API changes. For example, in Cargo.toml:
+``` toml
+taxonomy = "^0.4"
+```
+
+Based on an offline discussion with one Servo dev, it is worth to mention that Cargo might take 2 versions of the same crate. Consider this example:
+```toml
+taxonomy = "^0.4"
+```
+```toml
+taxonomy = "0.4.1"
+```
+If the most recent version of taxonomy is 0.4.2, Cargo will pick that one for the first build and pick 0.4.1 for the second one. Cargo won't pick the version that matches every constraint (which should have been 0.4.1). Therefore, we might build taxonomy twice, just like with master.
+
+Likewise, if somebody inadvertently breaks the API without increasing the major version (v0.5, in the example), we will break the other dependency. Thenceforth, we will need to run the tests of every actor using that dependency.
+
+#### Automate revisions bumps
+
+We could write a bumper script that will create a PR for each repo that depends on a bottom-level dependency and automatically lands it once all the PRs are green and an r+ was given on the initial PR. This seems to be the hardest solution in terms of initial set up. We will also rely more on Github than we currently do.
+
+### Have one repo with one single crate
+
+Bringing back taxonomy, thinkerbell and the adapters in foxbox will give a convenient way to find integration errors, as all the tests can be run locally without any particular inter-repository configuration.
+
+One major drawback would be the compile time. For more details about crate compile time, see [fxbox/foxbox#113](https://github.com/fxbox/foxbox/issues/113).
+
+### Have one main repo with many crates
+
+Everything that is tightly coupled or subject to heavy changes will live in the main repo. Other dependencies will remain in other repos. As of today, the repos which are candidates to be in the main repo are:
+* foxbox (aka the main repo)
+* openzwave-adapter
+* thinkerbell
+* taxonomy
+
+`cargo build` in the root directory will spot compile errors. A script to run `cargo test` in each submodule will be needed. If one would like to build/test only one module, we just need to cd to the right folder. Nonetheless, our CI jobs will likely take longer as more tests will need to be run in the same Travis job.
+
+Even if they are in the same repo, we can publish module on crates.io. E.g: [regex](https://github.com/rust-lang-nursery/regex) lives in the same repo as [regex_macros](https://github.com/rust-lang-nursery/regex/tree/master/regex_macros) and both are on crates.io (see [here](https://crates.io/crates/regex) and [there](https://crates.io/crates/regex_macros)).
+
+Foxbox would be the main entry point for contributors and we could describe the whole flow there.
```

Comment:
Additionally, we could move the current adapters to a distinct crate each, without hurting development.